pkg/webhook/vlanconfig/mutator.go

```diff
@@ -78,24 +80,31 @@ func getCnLabelPatch(v *networkv1.VlanConfig) types.Patch {
 }
 
 func (v *vlanConfigMutator) matchNodes(vc *networkv1.VlanConfig) (types.Patch, error) {
-	selector, err := utils.NewSelector(vc.Spec.NodeSelector)
-	if err != nil {
-		return nil, err
-	}
-	nodes, err := v.nodeCache.List(selector)
+	nodes, err := v.nodeCache.List(labels.Set(vc.Spec.NodeSelector).AsSelector())
 	if err != nil {
 		return nil, err
 	}
-
-	lenOfNodes := len(nodes)
-	if lenOfNodes == 0 {
+	if len(nodes) == 0 {
 		return nil, nil
 	}
 
-	matchedNodes := make([]string, lenOfNodes)
-	for i, node := range nodes {
-		matchedNodes[i] = node.Name
+	matchedNodes, pausedNodes := make([]string, 0, len(nodes)), make([]string, 0, len(nodes))
+	for _, node := range nodes {
+		vsName := utils.Name("", vc.Spec.ClusterNetwork, node.Name)
+		if vs, err := v.vsCache.Get(vsName); err != nil && !apierrors.IsNotFound(err) {
+			return nil, err
+		} else if err == nil && vs.Status.VlanConfig != vc.Name {
+			// The vlanconfig is found means a vlanconfig with the same clusternetwork
+			// has been taken effect on this node. We have to pause the new vlanconfig.
+			pausedNodes = append(pausedNodes, node.Name)
+		}
+		matchedNodes = append(matchedNodes, node.Name)
 	}
+
+	return matchedNodesToPatch(vc, matchedNodes, pausedNodes)
```

Comment:
instead of patching the spec would it be easier to add the check on the validator and reject to create directly?

Reply:
Yes. It's logically more direct.